src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -91,3 +92,23 @@ def _parse_asn1_time(self, asn1_time):
             )
         ).decode("ascii")
         return datetime.datetime.strptime(time, "%Y%m%d%H%M%SZ")
+
+    @property
+    def signature_hash_algorithm(self):
+        oid = self._signature_algorithm()
+        try:
+            return x509._SIG_OIDS_TO_HASH[oid.dotted_string]
+        except KeyError:
+            raise UnsupportedAlgorithm(
+                "Signature algorithm {0} not recognized".format(oid)
+            )
+
+    def _signature_algorithm(self):
+        buf_len = 50
+        buf = self._backend._ffi.new("char[]", buf_len)
+        res = self._backend._lib.OBJ_obj2txt(
+            buf, buf_len, self._x509.sig_alg.algorithm, 1
+        )
+        assert res <= 50 and res > 0
+        oid = self._backend._ffi.buffer(buf, res)[:].decode()
+        return x509.ObjectIdentifier(oid)
```

Comment:
If we're not going to make the keys be `ObjectIdentifier` instances, is there any reason to create an instance here?